contracts/src/tags/schema.ts

```diff
@@ -0,0 +1,198 @@
+  ),
+});
+
+export const superTagIdQuerySchema = z.object({
+  superTagId: positiveInt.openapi({
+    description: '슈퍼 태그의 id',
+    example: 1,
+  }),
+});
+
+export const subTagResponseSchema = z.object({
+  id: positiveInt.openapi({
+    description: '태그 고유 id',
+    example: 1,
+  }),
+  login: z.string().openapi({
+    description: '태그를 작성한 카뎃의 닉네임',
+    example: 'yena',
+  }),
+  content: z.string().openapi({
+    description: '서브/디폴트 태그의 내용',
+    example: 'yena가_추천하는',
+  }),
+});
+
+export const tagsOfBookResponseSchema = z.object({
+  items: z.array(
+    z.object({
+      id: positiveInt.openapi({
+        description: '태그 고유 id',
+        example: 1,
+      }),
+      login: z.string().openapi({
+        description: '태그를 작성한 카뎃의 닉네임',
+        example: 'yena',
+      }),
+      content: z.string().openapi({
+        description: '슈퍼/디폴트 태그의 내용',
+        example: 'yena가_추천하는',
+      }),
+      type: z.enum(['super', 'default']).openapi({
+        description: '태그의 타입. 슈퍼 태그는 super, 디폴트 태그는 default',
+        example: 'super',
+      }),
+      count: positiveInt.openapi({
+        description: '슈퍼 태그에 속한 서브 태그의 개수. 디폴트 태그는 0',
+        example: 1,
+      }),
+    }),
+  ),
+});
+
+export const modifySuperTagBodySchema = z.object({
+  id: positiveInt.openapi({
+    description: '수정할 슈퍼 태그의 id',
+    example: 1,
+  }),
+  content: z.string().openapi({
+    description: '수정할 슈퍼 태그의 내용',
+    example: '1서클_추천_책',
+  }),
+});
+
+export const modifyTagResponseSchema = z.literal('success');
+
+export const incorrectTagFormatSchema = mkErrorMessageSchema('INCORRECT_TAG_FORMAT')
+  .describe('태그 형식이 올바르지 않습니다.');
+
+export const alreadyExistTagSchema = mkErrorMessageSchema('ALREADY_EXIST_TAG')
+  .describe('이미 존재하는 태그입니다.');
+
+export const defaultTagCannotBeModifiedSchema = mkErrorMessageSchema('DEFAULT_TAG_CANNOT_BE_MODIFIED')
+  .describe('디폴트 태그는 수정할 수 없습니다.');
+
+export const modifySubTagBodySchema = z.object({
+  id: positiveInt.openapi({
+    description: '수정할 서브 태그의 id',
+    example: 1,
+  }),
+  content: z.string().openapi({
+    description: '수정할 서브 태그의 내용',
+    example: 'yena가_추천하는',
+  }),
+  visibility: z.enum(['public', 'private']).openapi({
+    description: '태그의 공개 여부. 공개는 public, 비공개는 private',
+    example: 'private',
+  }),
+});
+
+export const NoAuthorityToModifyTagSchema = mkErrorMessageSchema('NO_AUTHORITY_TO_MODIFY_TAG')
+  .describe('태그를 수정할 권한이 없습니다.');
+
+export const mergeTagsBodySchema = z.object({
+  superTagId: positiveInt.nullable().openapi({
+    description: '병합할 슈퍼 태그의 id. null이면 디폴트 태그로 병합됨을 의미한다.',
+    example: 1,
+  }),
+  subTagIds: z.array(positiveInt).openapi({
+    description: '병합할 서브 태그의 id 목록',
+    example: [1, 2, 3],
+  }),
+});
+
+export const invalidTagIdSchema = mkErrorMessageSchema('INVALID_TAG_ID')
+  .describe('태그 id가 올바르지 않습니다.');
+
+export const createTagBodySchema = z.object({
+  bookInfoId: positiveInt.openapi({
+    description: '태그를 등록할 도서의 info id',
+    example: 1,
+  }),
```

Comment:
```suggestion
  bookInfoId: bookInfoIdSchema,
```

https://github.com/jiphyeonjeon-42/backend/blob/112e19ac2f6ef7266f8acb2be9582f66f120878b/contracts/src/shared.ts#L7

`bookInfoIdSchema`와 같이 기존에 있던 스키마를 재활용하면 코드 중복을 줄일 수 있을 것 같아요.